qrhl/qrhl.el

```diff
@@ -0,0 +1,115 @@
+  "Finds the period-terminated command starting at the point (and moves to its end).
+   Returns t if this worked."
+  (let ((pos 
+	 (save-excursion 
+	   (progn
+	    (while (or
+	              ; skip forward over regular chars, period with non-white, quoted string
+		    (qrhl-forward-regex "\\([^.{(\"]+\\|\\.[^ \t\n]\\|\"\\([^\"]+\\)\"\\)")
+		    (and (looking-at "[{(]") (forward-list))
+		    ))
+	    (and (qrhl-forward-regex "\\.") (point))
+	    ))))
+    (princ pos)
+    (and pos (goto-char pos) t)))
+
+(defun qrhl-parse-focus-command ()
+  (and (looking-at qrhl-focus-cmd-regexp)
+       (goto-char (match-end 0))))
+
+(defun qrhl-proof-script-parse-function ()
+  "Finds the command/comment starting at the point"
+  (or (and (qrhl-forward-regex "#[^\n]*\n") 'comment)
+      (and (qrhl-parse-focus-command) 'cmd)
+      (and (qrhl-parse-regular-command) 'cmd)))
+
+(proof-easy-config 'qrhl "qRHL"
+		   proof-prog-name (concat qrhl-home "bin/qrhl")
+		   ; We need to give some option here, otherwise proof-prog-name is interpreted
+		   ; as a shell command which leads to problems if the path contains spaces
+		   ; (see the documentation for proof-prog-name)
+		   qrhl-prog-args '("--emacs")
+		   ;proof-script-command-end-regexp "\\.[ \t]*$"
+		   proof-script-parse-function 'qrhl-proof-script-parse-function
+		   proof-shell-annotated-prompt-regexp "^\\(\\.\\.\\.\\|qrhl\\)> "
+		   ;proof-script-comment-start-regexp "#"
+		   ;proof-script-comment-end "\n"
+		   proof-shell-error-regexp "^\\(\\[ERROR\\]\\|Exception\\)"
+		   proof-undo-n-times-cmd "undo %s."
+		   proof-find-and-forget-fn 'qrhl-find-and-forget
+		   proof-shell-start-goals-regexp "^[0-9]+ subgoals:\\|^Goal:\\|^No current goal\\.\\|^In cheat mode\\.\\|^No focused goals (use "
+		   proof-shell-proof-completed-regexp "^No current goal.$"
+		   proof-shell-eager-annotation-start "\\*\\*\\* "
+		   proof-shell-eager-annotation-start-length 4
+		   proof-no-fully-processed-buffer t
+		   proof-shell-filename-escapes '(("\\\\" . "\\\\") ("\"" . "\\\""))
+		   proof-shell-cd-cmd "changeDirectory \"%s\"."
+		   proof-save-command-regexp "^adfuaisdfaoidsfasd" ; ProofGeneral produces warning when this is not set. But we don't want goal/save commands to be recognized because that makes ProofGeneral do an atomic undo.
+		   )
+
+
+
+
+; buttoning functions follow https://superuser.com/a/331896/748969
+(define-button-type 'qrhl-find-file-button
+  'follow-link t
+  'action #'qrhl-find-file-button)
+
+(defun qrhl-find-file-button (button)
+  (find-file (buffer-substring (button-start button) (button-end button))))
+
+(defun qrhl-buttonize-buffer ()
+ "turn all include's into clickable buttons"
+ (interactive)
+ (remove-overlays)
+ (save-excursion
+  (goto-char (point-min))
+  (while (re-search-forward "include\s*\"\\([^\"]+\\)\"\s*\\." nil t)
+   (make-button (match-beginning 1) (match-end 1) :type 'qrhl-find-file-button))))
+
+
+
+
+(add-hook 'qrhl-mode-hook
```

Comment:
I removed part of the hardcoded stuff (0eb165d). 
Still hardcoded:
* The language environment (see the discussion above)
* `electric-indent-mode`: Since qrhl-mode does not provide a sensible indentation configuration, electric indent only leads to confusion, even for users who generally want electric indent. If I add a better indentation, I will remove this so that it defaults to global preferences.
* `qrhl-buttonize-buffer`: This makes `include "blabla".` commands clickable. Not a personal setting.

(I'm marking this resolved but feel free to reopen.)